agent/pkg/k8s/proxy.go

```diff
@@ -0,0 +1,21 @@
+package k8s
+
+// GetProxyLogLevel retrieves the proxy log level for a particular pod
+func (c *Client) GetProxyLogLevel(podName, namespace string) (string, error) {
+	return "info", nil
+}
+
+// SetProxyLogLevel sets the proxy log level for a particular pod
+func (c *Client) SetProxyLogLevel(podName, namespace, logLevel string) error {
+	return nil
+}
+
+// GetProxyLogs retrieves the proxy logs of a pod
+func (c *Client) GetProxyLogs(podName, namespace string) (string, error) {
+	return "here are some proxy logs", nil
+}
+
+// GetPrometheusScrape retrieves the raw prom scrape from the proxy of a pod
+func (c *Client) GetPrometheusScrape(podName, namespace string) (string, error) {
+	return "here are some prom metrics", nil
+}
```

Comment:
For hitting the endpoints, I think we can get away with just the pod DNS, but if that does not work we will most likely need to get the IP addr.  I was just thinking that this is the right way to put them since we are interacting with pods that are on the cluster even if we do not directly use the k8s api. Wrt to getting the actual logs, yes I was planning to use the go client to do that.

Reply:
i had mostly envisioned `k8s.Client` as a client for the k8s api, but this is not a blocker. as we move forward with implementation, see where it feels most ergonomic to put a `/metrics` scrape.